### PR TITLE
chore(Docker): add docker build to run tests

### PR DESCRIPTION
**Pull request by LuisFros**
[sc-71067]

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RUN npm i
+
+# Run ci checks
+RUN npm run ci-audit
+
+RUN npm run ci-mocha
+
+RUN npm run ci-eslint
+
+RUN npm test
```
Comment (mathiastj) [inline Dockerfile:27]: Any reason for running both `ci-mocha` and `test`?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+ARG NODE_VERSION=14.x
+
+FROM gcr.io/connectedcars-staging/node-builder.master:$NODE_VERSION as builder
+
+ARG SSH_KEY_PASSWORD
+ARG NPM_TOKEN
```
Comment (mathiastj) [inline Dockerfile:6]: I don't think you need either of those, since this is not using any private packages.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ARG COMMIT_SHA=master
+
+WORKDIR /app
+
+RUN apt-get update
```
Comment (mathiastj) [inline Dockerfile:11]: I don't see the need for `apt-get update` when it's just for running the tests.

File: Dockerfile (hunk 1 above)
Comment (LuisFros) [reply Dockerfile:27]: Its so that the build fails if the tests fails. Like in https://github.com/connectedcars/node-setup/blob/master/Dockerfile

Comment (LuisFros) [reply Dockerfile:27]: I will add that comment message

Comment (mathiastj) [reply Dockerfile]: You should be able to add `--hard-fail` to ci-mocha, introduced to checks here: https://github.com/connectedcars/node-test/pull/31

**Review by mathiastj — APPROVED** [review]

**Pull request merged**